clang/include/clang/Serialization/ASTReader.h

```diff
@@ -237,6 +237,17 @@ class ASTReaderListener {
     return true;
   }
 
+  /// Overloaded member function of \c visitInputFile that should
+  /// be defined when the input file contains both the virtual and external
+  /// paths, for example when deserializing input files from AST files.
+  ///
+  /// \returns true to continue receiving the next input file, false to stop.
+  virtual bool visitInputFile(StringRef FilenameAsRequested,
+                              StringRef ExternalFilename, bool isSystem,
```

Comment:
Nit: I think the `ExternalFilename` name is a bit misleading. That's only a concept in `llvm::RedirectingFileSystem` and even there it's controlled by a setting. In Clang, we'd call this just `Filename`.

Reply:
Is it worth documenting whats the intended difference between `Filename` vs `FilenameAsRequested`? Or is the docstring already enough?

Reply:
I'd probably just defer to other definitions of `Filename` vs `FilenameAsRequested`:

```
   /// Overloaded member function of \c visitInputFile that should
   /// be defined when the listener cares about the distinction between
   /// the file name and name-as-requested.
```